src/index.ts

```diff
@@ -228,7 +228,12 @@ export interface Purpose {
   categories?: PurposeCategory[]
   tcfType?: string
   tcfId?: string
-  canonicalPurposeCode: string
+  legalBasisName?: string
+  legalBasisDescription?: string
+}
+
+export interface CanonicalPurpose {
+  purposeCodes?: string[]
```

Comment:
these are purposeCodes mapped to this canonical purpose?

Reply:
or canonical purpose codes?

Reply:
yes - naming suggestion?